### PR TITLE
Make repo compliant

### PR DESCRIPTION
**Pull request by RaphaelVogel**
**What this PR does / why we need it**:
* Updated `CONTRIBUTING.md` file
* Created/Updated `LICENSE` file
* Created/Updated `.reuse/dep5` file
* Created/Updated `LICENSES` folder
* Deleted `NOTICE.md` file
* Added badge for reuse compliance in `README.md` file and registered repo with REUSE API
* License/Copyright header replacement for 'go' files
  * hack/tools.go

### CODE REVIEW

File: .reuse/dep5 [hunk 1]
```diff
@@ -2,65 +2,294 @@ Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
 Upstream-Name: Gardener ext-authz-server
 Upstream-Contact: The Gardener project <gardener@googlegroups.com>
 Source: https://github.com/gardener/ext-authz-server
+
```
Comment (ScheererJ) [inline .reuse/dep5:5]: Why did you add a new line? Are you paid by lines of code?

File: .reuse/dep5 [hunk 2]
```diff
@@ -2,65 +2,294 @@ Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
...
 License: Apache-2.0
 
 # --------------------------------------------------
 # documentation
 
-Files:
-  *.md
-Copyright: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+Files: *.md
+Copyright: 2017-2024 SAP SE or an SAP affiliate company and Gardener contributors
```
Comment (ScheererJ) [inline .reuse/dep5:17]: Why do we use a different license for our markdown files compared with the source code?

File: .reuse/dep5 [hunk 3]
```diff
@@ -2,65 +2,294 @@ Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
...
+# third-party
+
+# --- copied source code ---
+# Files:
+# Copyright:
+# License:
+
+# --- vendor folder dependencies ---
+Files: vendor/github.com/google/gnostic/*
+Copyright: 2024 github.com/google/gnostic contributors.
```
Comment (ScheererJ) [inline .reuse/dep5:30]: Could you please instead of updating the vendor licenses please create a pull request for vendor folder removal?

/hold

**Review by DockToFuture — APPROVED** [review]
/lgtm

File: LICENSES/BSD-3-Clause.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) <year> <owner>. All rights reserved.
+Copyright (c) <year> <owner>.
```
Comment (ScheererJ) [inline LICENSES/BSD-3-Clause.txt:1]: Why are the rights no longer reserved?

**Review by ScheererJ — CHANGES_REQUESTED** [review]

File: LICENSES/BSD-3-Clause.txt (hunk 4 above)
Comment (axel7born) [reply LICENSES/BSD-3-Clause.txt:1]: The requirement to add the "all rights reserved" notice became essentially obsolete on August 23, 2000, when Nicaragua became the final member of the Buenos Aires Convention to also become a signatory to the Berne Convention.
https://en.wikipedia.org/wiki/All_rights_reserved

File: .reuse/dep5 (hunk 3 above)
Comment (ScheererJ) [reply .reuse/dep5:30]: @RaphaelVogel Please update this pull request as https://github.com/gardener/ext-authz-server/pull/28 has been merged!

**Pull request closed without merging**